mllib/src/test/scala/org/apache/spark/ml/feature/HashingTFSuite.scala

```diff
@@ -52,6 +52,27 @@ class HashingTFSuite extends SparkFunSuite with MLlibTestSparkContext with Defau
     assert(features ~== expected absTol 1e-14)
   }
 
+  test("applying binary term freqs") {
+    val df = sqlContext.createDataFrame(Seq(
+      (0, "a a b c c c".split(" ").toSeq)
+    )).toDF("id", "words")
+    val n = 100
+    val hashingTF = new HashingTF()
+        .setInputCol("words")
+        .setOutputCol("features")
+        .setNumFeatures(n)
+        .setBinary(true)
+    val output = hashingTF.transform(df)
+    val attrGroup = AttributeGroup.fromStructField(output.schema("features"))
+    require(attrGroup.numAttributes === Some(n))
+    val features = output.select("features").first().getAs[Vector](0)
+    // Assume perfect hash on "a", "b", "c".
+    def idx(any: Any): Int = Utils.nonNegativeMod(any.##, n)
```

Comment:
Do you mean expose "indexOf" in ml to match mllib?
On Tue, 22 Mar 2016 at 19:27, Bryan Cutler <notifications@github.com> wrote:

> In mllib/src/test/scala/org/apache/spark/ml/feature/HashingTFSuite.scala
> <https://github.com/apache/spark/pull/11832#discussion_r57030239>:
>
> > +  test("applying binary term freqs") {
> > +    val df = sqlContext.createDataFrame(Seq(
> > +      (0, "a a b c c c".split(" ").toSeq)
> > +    )).toDF("id", "words")
> > +    val n = 100
> > +    val hashingTF = new HashingTF()
> > +        .setInputCol("words")
> > +        .setOutputCol("features")
> > +        .setNumFeatures(n)
> > +        .setBinary(true)
> > +    val output = hashingTF.transform(df)
> > +    val attrGroup = AttributeGroup.fromStructField(output.schema("features"))
> > +    require(attrGroup.numAttributes === Some(n))
> > +    val features = output.select("features").first().getAs[Vector](0)
> > +    // Assume perfect hash on "a", "b", "c".
> > +    def idx(any: Any): Int = Utils.nonNegativeMod(any.##, n)
>
> It actually kind of bugged me that these tests are defining their own
> index calculations, even though it is just a mod. Since ml.HashingTF
> doesn't have this available, I'll try out a wrapping the mllib function
> being used, which I think is more correct.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/11832/files/19849fe49d1f0be62c9bf54f1312d8cfc457f003#r57030239>
>